File: backend/order-service/src/main/java/com/mall/order/entity/Order.java
package com.mall.order.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

@Data
@TableName("`order`")
public class Order {
    @TableId(type = IdType.AUTO)
    private Long id;
    private Long userId;
    private String orderNo;
    private BigDecimal totalAmount;
    private Integer status; // 0=pending_payment, 1=pending_delivery, 2=pending_receipt, 3=completed, 4=cancelled, 5=refunding
    private String paymentMethod;
    private LocalDateTime paymentTime;
    private LocalDateTime deliveryTime;
    private LocalDateTime receiveTime;
    private String receiverName;
    private String receiverPhone;
    private String receiverAddress;
    private LocalDateTime createdTime;
    private LocalDateTime updatedTime;
    @TableField(exist = false)
    private List<OrderItem> items;
}

File: frontend/src/admin/api/order.ts
import request from './request'
import type { PageResult } from './product'

export interface OrderItem {
  id: number
  orderId: number
  productId: number
  productName: string
  productImage: string
  specInfo: string | null
  price: number
  quantity: number
  subtotal: number
}

export interface Order {
  id: number
  userId: number
  orderNo: string
  totalAmount: number
  status: number
  paymentMethod: string | null
  paymentTime: string | null
  deliveryTime: string | null
  receiveTime: string | null
  receiverName: string
  receiverPhone: string
  receiverAddress: string
  createdTime: string
  updatedTime: string
  items: OrderItem[] | null
}

export function getAdminOrderList(params: {
  status?: number
  page?: number
  size?: number
}) {
  return request.get<any, PageResult<Order>>('/order/admin/list', { params })
}

export function updateOrderStatus(id: number, status: number) {
  return request.put<void>('/order/admin/status', null, { params: { id, status } })
}
